config/src/config/secure_config.rs

```diff
@@ -126,3 +168,79 @@ impl OnDiskStorageConfig {
         self.data_dir = data_dir;
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::io::Write;
+
+    #[test]
+    fn test_token_config_parsing() {
```

Comment:
I'd maybe split this into two tests? e.g., one called "test_from_token_config_parsing()" and one for test_from_disk_parsing()"?